packages/autoupdate-polling/autoupdate-polling-client.js

```diff
@@ -0,0 +1,160 @@
+// A "hard code push" occurs when the running client version is not in
+// the set of acceptable client versions (or the server updates the
+// collection, there is a published client version marked `current` and
+// the running client version is no longer in the set).
+//
+// When the `reload` package is loaded, a hard code push causes
+// the browser to reload, so that it will load the latest client
+// version from the server.
+//
+// A "soft code push" represents the situation when the running client
+// version is in the set of acceptable versions, but there is a newer
+// version available on the server.
+
+// The client version of the client code currently running in the
+// browser.
+import { ClientVersions } from "./client-versions.js";
+
+const clientArch = Meteor.isCordova ? "web.cordova" :
+  Meteor.isModern ? "web.browser" : "web.browser.legacy";
+
+const autoupdateVersions =
+  ((__meteor_runtime_config__.autoupdate || {}).versions || {})[clientArch] || {
+    version: "unknown",
+    versionRefreshable: "unknown",
+    versionNonRefreshable: "unknown",
+    assets: [],
+  };
+
+export const AutoupdatePolling = {};
+
+// Stores acceptable client versions.
+const clientVersions =
+  AutoupdatePolling._clientVersions = // Used by a self-test.
+  new ClientVersions();
+
+AutoupdatePolling.newClientAvailable = function () {
+  return clientVersions.newClientAvailable(
+    clientArch,
+    ["versionRefreshable", "versionNonRefreshable"],
+    autoupdateVersions
+  );
+};
+
+let lastInterval;
+
+AutoupdatePolling._pollingSubscribe = (delay) => {
+  Meteor.clearInterval(lastInterval);
+  lastInterval = Meteor.setInterval(() => {
+
+    var xhr = new XMLHttpRequest();
+    xhr.addEventListener("load", function (evt ) {
+      const mostRecentVersion = JSON.parse(evt.target.response);
+      clientVersions.set(clientArch, { ...mostRecentVersion });
+    });
+    xhr.open("GET", `/meteor_autoupdate_polling_clientVersions?arch=${clientArch}`);
+    xhr.send();
+  }, delay);
+};
+
+if (!Meteor.isProduction || Meteor.settings.public.autoupdate_polling_time)
```

Comment:
Better to use `Meteor.settings.public?.autoupdate_polling_time` to prevent errors when public settings is not set.

Reply:
I think this won't be necessary because `Meteor.settings.public` is guaranteed to always be defined according to [the docs](https://docs.meteor.com/api/core.html#Meteor-settings), even if there are no settings specified.